GitUI/CommandsDialogs/FormBrowse.cs

```diff
@@ -1635,18 +1637,36 @@ private void ChangeWorkingDir(string path)
                 return;
             }
 
-            DialogResult dialogResult = MessageBox.Show(this, _directoryIsNotAValidRepository.Text,
-                                                        _directoryIsNotAValidRepositoryCaption.Text,
-                                                        MessageBoxButtons.YesNo,
-                                                        MessageBoxIcon.Exclamation,
-                                                        MessageBoxDefaultButton.Button1);
+            int invalidPathCount = ThreadHelper.JoinableTaskFactory.Run(() => RepositoryHistoryManager.Locals.LoadRecentHistoryAsync()).Count(repo => !GitModule.IsValidGitWorkingDir(repo.Path));
+            string commandButtons = _directoryIsNotAValidRepositoryRemoveSelectedRepoCommand.Text;
```

Comment:
The chosen variable poorly conveys the intent, it took me few minutes to realise it contains buttons' captions.
Let's rename it to something that is clearer, e.g. `commandButtonCaptions` or `commandButtonText`